problemi/boltzmannDistribution/boltzmannDistribution.cpp

```diff
@@ -0,0 +1,144 @@
+#include <TCanvas.h>
+#include <TF1.h>
+#include <TH1D.h>
+#include <TLegend.h>
+
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "boltzmann/boltzmann.h"
+
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include "doctest.h"
+
+void plotDistributionROOT(const std::vector<int> &interactions_list,
+                          int molecules) {
+  TCanvas *c1 = new TCanvas("c1", "Boltzmann Simulations", 1200, 800);
+  c1->Divide(std::ceil(interactions_list.size() / 2.), 2);
+  std::cout << "aa" << std::ceil(interactions_list.size() / 2) << std::endl;
+  int pad{1};
+  for (auto nInteractions : interactions_list) {
+    BoltzmannSimulator sim(molecules);
+    sim.run(nInteractions);
+
+    SimStats st{sim.getStats()};
+    std::cout << "Number of interactions=" << nInteractions
+              << " mean=" << st.mean << " stddev=" << st.stddev << std::endl;
+
+    c1->cd(pad);
+    std::string hname{"hE_" + std::to_string(nInteractions)};
+    TH1D *hE = new TH1D(
+        hname.c_str(),
+        ("Energy distribution interactions=" + std::to_string(nInteractions))
+            .c_str(),
+        80, 0, 10);
+    for (double e : sim.getEnergies()) hE->Fill(e);
+
+    hE->GetXaxis()->SetTitle("Energy");
+    hE->GetYaxis()->SetTitle("Count");
+    hE->SetLineColor(pad);
+    hE->Draw();
+
+    // Perform the exponential fit
+    if (nInteractions > molecules * 10) {
+      TF1 *fexp = new TF1("fexp", "expo", 0, 10);
+      hE->Fit(fexp, "RQ");
+      fexp->SetLineColor(2);
+      fexp->Draw("same");
+    }
+    pad++;
+  }
+
+  c1->SaveAs("problemi/boltzmannDistribution/boltzmann_distributions.png");
+}
+
+void plotDistributionSTDOUT(const std::vector<int> &interaction_list,
```

Comment:
```suggestion
void plot_distribution_stdout(const std::vector<int> &interaction_list,
```
ma non si potrebbe usare l'overload della funzione e avere un parametro `std::ostream&`, cosi' che si possa scrivere su stdout o su un ofstream?